tests/LSP6KeyManager/tests/ExecuteRelayCall.test.ts

```diff
@@ -0,0 +1,367 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { TargetContract, TargetContract__factory } from "../../../types";
+
+// constants
+import {
+  ALL_PERMISSIONS,
+  ERC725YKeys,
+  OPERATION_TYPES,
+  PERMISSIONS,
+} from "../../../constants";
+
+// helpers
+import { combinePermissions } from "../../utils/helpers";
+
+// setup
+import { LSP6TestContext } from "../../utils/context";
+import { setupKeyManager } from "../../utils/fixtures";
+import { provider } from "../../utils/helpers";
+
+export const shouldBehaveLikeExecuteRelayCall = (
+  buildContext: () => Promise<LSP6TestContext>
+) => {
+  let context: LSP6TestContext;
+
+  let signer: SignerWithAddress,
+    relayer: SignerWithAddress,
+    random: SignerWithAddress;
+  let targetContract: TargetContract;
+
+  before(async () => {
+    context = await buildContext();
+
+    signer = context.accounts[1];
+    relayer = context.accounts[2];
+    random = context.accounts[3];
+
+    targetContract = await new TargetContract__factory(
+      context.accounts[0]
+    ).deploy();
+
+    const permissionKeys = [
+      ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+        context.owner.address.substring(2),
+      ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+        signer.address.substring(2),
+    ];
+
+    const permissionsValues = [
+      ALL_PERMISSIONS,
+      combinePermissions(PERMISSIONS.CALL, PERMISSIONS.TRANSFERVALUE),
+    ];
+
+    await setupKeyManager(context, permissionKeys, permissionsValues);
+  });
+
+  describe("When testing executeRelayCall(..)", () => {
+    describe("When testing signed message", () => {
+      describe("When testing msg.value", () => {
+        describe("When sending more than the signed msg.value", () => {
+          it("should revert by recovering a non permissioned address", async () => {
+            let executeRelayCallPayload =
+              context.universalProfile.interface.encodeFunctionData("execute", [
+                OPERATION_TYPES.CALL,
+                random.address,
+                0,
+                "0x",
+              ]);
+
+            let latestNonce = await context.keyManager.callStatic.getNonce(
+              signer.address,
+              0
+            );
+
+            let valueToSign = 5;
+
+            const signedMessageParams = {
+              chainId: 31337, // HARDHAT_CHAINID
+              address: context.keyManager.address,
+              nonce: latestNonce,
+              Msgvalue: valueToSign,
```

Comment:
minor recommendation

```suggestion
              value: valueToSign,
```

or 

```suggestion
              msgValue: valueToSign,
```

Reply:
Will go with the second one to emphasize that this is the `msg.value` not the `value` of the `execute(..)` function